diff --git a/sam3/model/sam3_multiplex_tracking.py b/sam3/model/sam3_multiplex_tracking.py
index 35183a5..658a1cf 100644
--- a/sam3/model/sam3_multiplex_tracking.py
+++ b/sam3/model/sam3_multiplex_tracking.py
@@ -1241,8 +1241,5 @@ class Sam3MultiplexTracking(Sam3MultiplexBase):
     def _build_sam2_output(
         self, inference_state, frame_idx, refined_obj_id_to_mask=None
     ):
-        if not frame_idx in inference_state["cached_frame_outputs"]:
-            return {}
-
-        cached_outputs = inference_state["cached_frame_outputs"][frame_idx]
+        cached_outputs = inference_state["cached_frame_outputs"].get(frame_idx, {})
         obj_id_to_mask = cached_outputs.copy()
